src/components/Popups/ReplyField/ItemList.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { SyntheticEvent } from 'react';
+import DatalistItem from 'box-ui-elements/es/components/datalist-item';
+import noop from 'lodash/noop';
+import { Collaborator } from '../../../@types';
+
+export type Props = {
+    activeItemIndex?: number;
+    filterString?: string;
+    itemRow?: JSX.Element;
+    items: Collaborator[];
+    onSelect: (index: number, event: React.SyntheticEvent) => void;
+    setActiveItemIndex?: (index: number) => void;
+};
+
+const DefaultItemRow = ({ item, ...rest }: Collaborator): JSX.Element | null =>
+    item?.name ? (
+        <DatalistItem {...rest}>
+            <div className="ba-PopupList-name">{item.name}</div>
+            {'email' in item && <div className="ba-PopupList-email">{item.email}</div>}
+        </DatalistItem>
+    ) : null;
+
+const ItemList = ({
+    activeItemIndex = 0,
+    itemRow = <DefaultItemRow {...({} as Collaborator)} />,
+    items,
+    onSelect,
+    setActiveItemIndex = noop,
+    ...rest
+}: Props): JSX.Element => {
+    return (
+        <ul role="listbox" {...rest}>
+            {items.map((item, index) => {
+                if (!item) {
+                    return null;
+                }
+
+                const itemProps = {
+                    ...item,
+                    key: item.id,
+                    className: 'ba-PopupList-item',
+                    isActive: index === activeItemIndex,
+                    onClick: (event: SyntheticEvent) => {
+                        onSelect(index, event);
+                    },
+                    /* preventDefault on mousedown so blur doesn't happen before click */
+                    onMouseDown: (event: SyntheticEvent) => {
+                        event.preventDefault();
+                    },
+                    onMouseEnter: () => {
+                        setActiveItemIndex(index);
+                    },
+                };
+
+                return React.cloneElement(itemRow as JSX.Element, itemProps);
```

Comment:
Why is this cast needed? Can we add a truthiness check above, similar to `item`?